clippy_lints/src/lifetimes.rs

```diff
@@ -394,7 +394,7 @@ fn non_elidable_self_type<'tcx>(cx: &LateContext<'tcx>, func: &FnDecl<'tcx>, ide
         let mut visitor = RefVisitor::new(cx);
         visitor.visit_ty_unambig(self_ty);
 
-        !visitor.all_lts().is_empty()
+        visitor.all_lts().next().is_some()
```

Comment:
I don't understand how this is better than `is_empty()` here. You could also use `impl Iterator<Item = Lifettime> + ExactSizeIterator`.

Reply:
no collection? only need one item?